modules/gateway/upnp.go

```diff
@@ -61,7 +66,7 @@ func (g *Gateway) learnHostname(port string) {
 		return
 	}
 
-	addr := modules.NetAddress(net.JoinHostPort(host, port))
+	addr := modules.NetAddress(net.JoinHostPort(host, g.port))
```

Comment:
Despite its convenience, I think I'd rather not have an `immutable` convention, since it would be a lie. Immutable fields seem to be pretty rare anyway; the only other one that comes to mind is `persistDir`.

Also, consider that during testing, you may want to mutate the "immutable" field. If locks are not used when accessing the field, then it becomes impossible to mutate them without triggering a data race.